pkg/virt-controller/watch/vmi/lifecycle.go

```diff
@@ -966,6 +969,56 @@ func (c *Controller) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 	}
 }
 
+func (c *Controller) syncAutoMigration(vmi *virtv1.VirtualMachineInstance) {
+	const pendingMigrationReEvalPeriod = 10 * time.Second
+
+	if migrations.IsMigrating(vmi) {
+		return
+	}
+
+	result := c.netMigrationEvaluator.Evaluate(vmi)
+
+	cm := controller.NewVirtualMachineInstanceConditionManager()
+	existingCondition := cm.GetCondition(vmi, virtv1.VirtualMachineInstanceMigrationRequired)
+
+	switch {
+	case result == k8sv1.ConditionUnknown && existingCondition == nil:
+		return
+	case result == k8sv1.ConditionUnknown && existingCondition != nil:
+		cm.RemoveCondition(vmi, virtv1.VirtualMachineInstanceMigrationRequired)
+		return
+	}
+
+	if existingCondition != nil && existingCondition.Status == k8sv1.ConditionTrue {
```

Comment:
once `existingCondition.Status == k8sv1.ConditionTrue`
it can't go back to `result == k8sv1.ConditionUnknown` right ?

if so, would it make sense to move it before the switch ?
(i do understand the current method is safer wrt to its own functionality, but if we don't expect such case, maybe it is better to be explicit and not keep potential unpredictable states)